### PR TITLE
Fix Python Agent running on GPT-4

### PR DESCRIPTION
**Pull request by durdn**
I went absolutely insane trying to write an Agent with multiple tools because of the exception below. I finally managed to track down the problem and reproduce it and fix it on the [Python Agent example](https://python.langchain.com/en/latest/modules/agents/toolkits/examples/python.html).

When running the vanilla example on `gpt-4` this is what you would get:

outputparserexception

I had to tweak the `mrkl` prompts to get things to work both on `gpt-3.5` and `gpt-4`.

### CODE REVIEW

**Comment by durdn** [comment]
Also note that I got a different answer on the fib number question depending on the `gpt` version. I guess it depends on where you start counting the Fibonacci numbers, if you include the initial `0`, then the 10th fib is `34`.

File: langchain/agents/mrkl/prompt.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 Question: the input question you must answer
 Thought: you should always think about what to do
-Action: the action to take, should be one of [{tool_names}]
+Action: the action to take, should be one of [{tool_names}, None]
```
Comment (hwchase17) [inline langchain/agents/mrkl/prompt.py:7]: hmmm why is this neccesary?

Comment (durdn) [reply langchain/agents/mrkl/prompt.py:7]: if I use `gpt-3.5` without that addition the notebook fails complaining that `Python REPL` tool does not exist. I think there might be a problem with parsing a `single` tool. I could try to debug further why that happens.

Comment (hwchase17) [reply langchain/agents/mrkl/prompt.py:7]: yeah that would be interesting to look into. seems a bit janky to add None

Comment (durdn) [reply langchain/agents/mrkl/prompt.py:7]: Fair enough, I'll spend more time debugging that behavior and report

Comment (durdn) [reply langchain/agents/mrkl/prompt.py:7]: I retested it and if I remove that ugly ` None` section, the agent can't find the `Python REPL` tool and goes in a loop with:

```Observation: [Python REPL] is not a valid tool, try another one.```

Now let's see if I can debug why things go wrong.

**Comment by Cobryis** [comment]
This is how I fixed the LLM output issue: https://github.com/hwchase17/langchain/pull/3480

**Comment by durdn** [comment]
There's been lots of work to make the output parser more robust so possibly these suggestions don't make sense anymore. Closing this for now.

**Pull request closed without merging**